contracts/contracts/strategies/BridgedWOETHStrategy.sol

```diff
@@ -0,0 +1,303 @@
+
+        // Do some checks
+        if (lastOraclePrice > 0) {
+            // Make sure the value only goes up
+            require(lastOraclePrice <= oraclePrice128, "Negative wOETH yield");
+
+            // And that it's within the bounds.
+            require(
+                // oraclePrice <= (lastOraclePrice * (1 + maxPriceDiffBps))
+                oraclePrice128 <=
+                    ((lastOraclePrice * (1e4 + uint128(maxPriceDiffBps))) /
+                        1e4),
+                "Price diff beyond threshold"
+            );
+        }
+
+        emit WOETHPriceUpdated(lastOraclePrice, oraclePrice128);
+
+        // Store the price
+        lastOraclePrice = oraclePrice128;
+
+        return oraclePrice;
+    }
+
+    /**
+     * @dev Wrapper for _updateWOETHOraclePrice with nonReentrant flag
+     * @return The latest price of wOETH from Oracle
+     */
+    function updateWOETHOraclePrice() external nonReentrant returns (uint256) {
+        return _updateWOETHOraclePrice();
+    }
+
+    /**
+     * @dev Computes & returns the value of given wOETH in WETH
+     * @param woethAmount Amount of wOETH
+     * @return Value of wOETH in WETH (using the last stored oracle price)
+     */
+    function getBridgedWOETHValue(uint256 woethAmount)
+        public
+        view
+        returns (uint256)
+    {
+        return (woethAmount * lastOraclePrice) / 1 ether;
+    }
+
+    /**
+     * @dev Takes in bridged wOETH and mints & returns
+     *      equivalent amount of OETHb.
+     * @param woethAmount Amount of bridged wOETH to transfer in
+     */
+    function depositBridgedWOETH(uint256 woethAmount)
+        external
+        onlyGovernorOrStrategist
+        nonReentrant
+    {
+        // Update wOETH price
+        uint256 oraclePrice = _updateWOETHOraclePrice();
+
+        // Figure out how much they are worth
+        uint256 oethToMint = (woethAmount * oraclePrice) / 1 ether;
+
+        // There's no pToken, however, it just uses WOETH address in the event
+        emit Deposit(address(weth), address(bridgedWOETH), oethToMint);
+
+        // Mint OETHb tokens and transfer it to the caller
+        IVault(vaultAddress).mintForStrategy(oethToMint);
+
+        // Transfer out minted OETHb
+        oethb.transfer(msg.sender, oethToMint);
+
+        // Transfer in all bridged wOETH tokens
+        bridgedWOETH.transferFrom(msg.sender, address(this), woethAmount);
+    }
+
+    /**
+     * @dev Takes in OETHb and burns it and returns
+     *      equivalent amount of bridged wOETH.
+     * @param woethAmount Amount of bridged wOETH to get back
+     */
+    function withdrawBridgedWOETH(uint256 woethAmount)
+        external
+        onlyGovernorOrStrategist
+        nonReentrant
+    {
+        // Update wOETH price
+        uint256 oraclePrice = _updateWOETHOraclePrice();
+
+        // Figure out how much they are worth
+        uint256 oethToBurn = (woethAmount * oraclePrice) / 1 ether;
```

Comment:
🔵 This actually rounds against the protocol, in favor of the "user", since oethToBurn is being rounded down, and this then the amount that the "user" has to pay.  This would be acceptable here with the small error size and the limited amount that this rounding could affect us.

However, I think we might (or might not) want to change this function for another reason. Let's say the strategist has 1,000 superOETHb in their wallet that they want to use to withdraw and get `bridgedWOETH` back. As it is, they will have to calculate the exchange rate to request woethAmount, then leave off some for rounding. It would be more work than just "I have a 1,000 superOETHb, let's do a withdraw". For example, OUSD mints and redeems work where the quantity in the function argument is always denominated in token that the user is holding.

Now as it is written, it's better for withdrawing the full amount held in this strategy, but it feels like this would happen less often than partial withdraws.

I'm open on this one, whichever way you like.

Reply:
That's a good point. I thought about it while writing the code, checked other strategies, we always passed the same asset addresses in both `deposit` and `withdraw` (since Vault is the one calling it). But yeah, it makes more sense to specify amount of OETHb to burn in this strategy. Besides full withdrawal shouldn't be that complicated with OETHb amount as parameter (since you can always check the oracle for the latest price and use that to compute how much you wanna burn).

Reply:
Have changed it now